RecoLocalMuon/GEMSegment/plugins/ME0SegAlgoRU.cc

```diff
@@ -29,14 +29,16 @@ ME0SegAlgoRU::ME0SegAlgoRU(const edm::ParameterSet& ps)
 
 	allowWideSegments = ps.getParameter<bool>("allowWideSegments");
 	doCollisions = ps.getParameter<bool>("doCollisions");
+
 	stdParameters.maxChi2Additional = ps.getParameter<double>("maxChi2Additional");
 	stdParameters.maxChi2Prune      = ps.getParameter<double>("maxChi2Prune"     );
 	stdParameters.maxChi2GoodSeg    = ps.getParameter<double>("maxChi2GoodSeg"   ) ;
 	stdParameters.maxPhiSeeds       = ps.getParameter<double>("maxPhiSeeds"      ) ;
 	stdParameters.maxPhiAdditional  = ps.getParameter<double>("maxPhiAdditional" );
 	stdParameters.maxETASeeds       = ps.getParameter<double>("maxETASeeds"      );
 	stdParameters.maxTOFDiff        = ps.getParameter<double>("maxTOFDiff"      );
-	stdParameters.minNumberOfHits        = ps.getParameter<unsigned int>("minNumberOfHits"      );
+	stdParameters.requireCentralBX  = ps.getParameter<bool>("requireCentralBX"      );
+	stdParameters.minNumberOfHits   = ps.getParameter<unsigned int>("minNumberOfHits"      );
 	stdParameters.requireBeamConstr = true;

```

Comment:
@perrotta thanks for the information. I've looked at this page and don't see an application that will work for me. For example, I can have each algorithm have its own "ConfigurationDescriptions" that then is called on by the Producer's "fillDescription" function. But I don't see how it will resolve the problem of having a vector of ParameterSets with different fields. 

I think that what we would need to do is change the way the ParameterSet is structured, but would prefer to not to do that here. I don't know how close we are to the next pre-release but we would like for this to get in here. Also, this exact same structure is used in GEMSegmentProducer and CSCSegmentProducer....I would hope that we would have a solution that matches across all three classes.